src/libgit2/repository.c

```diff
@@ -582,10 +582,14 @@ static int validate_ownership_cb(const git_config_entry *entry, void *payload)
 		         strncmp(test_path, "//wsl.localhost/", strlen("//wsl.localhost/")) != 0)
 			test_path++;
 #endif
-
-		if (git_fs_path_prettify_dir(&data->tmp, test_path, NULL) == 0 &&
-		    strcmp(data->tmp.ptr, data->repo_path) == 0)
+		git_str_sets(&tmp, data->repo_path);
```

Comment:
We were passing the `git_str` here to avoid re-allocing a `git_str` on every call. We can just re-use the one we have. No need to undo that, I don't think.

Reply:
Also, you're looking at `data->repo_path`, not `test_path`, which we've mutated to handle WSL paths. We should keep that.